integration_tests/src/main/python/array_test.py

```diff
@@ -445,7 +445,7 @@ def q1(spark):
 
 @incompat
 @pytest.mark.parametrize('data_gen', no_neg_zero_all_basic_gens + decimal_gens, ids=idfn)
-@pytest.mark.skipif(is_before_spark_313() or is_spark_330_or_later() or is_databricks104_or_later(), reason="NaN equality is only handled in Spark 3.1.3+")
+@pytest.mark.skipif(is_before_spark_313() or is_spark_330(), reason="NaN equality is only handled in Spark 3.1.3+")
```

Comment:
```suggestion
@pytest.mark.skipif(is_before_spark_313() or spark_version() == '3.3.0', reason="NaN equality is only handled in Spark 3.1.3+")
```

Reply:
The issue is that version comparison functions are not as trivial for some cases like databricks. Consistency with the existing code base and the ability to change the implementation of the version comparisons without touching all the tests speak in favor of  the current PR implementation.

Reply:
I see the reasons for both approaches. I made the change because I think the function `is_spark_330()` may not be reused later for other cases. I am open to suggestions, thanks!

Reply:
Inconsistency increases the maintenance overhead. You can propose to change the current style in a dedicated issue https://github.com/NVIDIA/spark-rapids/blob/0f012e51129d4bd94ea5c0984fe85f9ea8f299b9/integration_tests/src/main/python/spark_session.py#L132-L160